### PR TITLE
Minimal DaVinci job run locally

### PR DESCRIPTION
**Pull request by alexpearce**
Addresses issue #7. Guides the student through creating an options file that will create an ntuple of D*+ candidates when run over a stripped DST.

### CODE REVIEW

File: 09-minimal-dv-job.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+>
+> A microDST (or µDST) is a smaller version of a DST.
+> Some stripping lines go to µDSTs, and some go to DSTs.
+> There are two things that need changing it our options file in order to have 
+> in work when it is used with a stripping line that goes to a µDST:
+>
+> 1. The `DecayTreeTuple.Inputs` attribute should start at the word
+>    `Phys`; and
+> 2. The `RootInTES` attribute on the `DaVinci` object to be set to 
+>    `/Event/STREAM`
```
Comment (betatim) [inline 09-minimal-dv-job.md]: Make this $STREAM or some such to indicate it is a variable, not the literal word STREAM

Comment (alexpearce) [inline 09-minimal-dv-job.md]: :+1:

File: 09-minimal-dv-job.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+```python
+from Configurables import DaVinci
+
+# Configure DaVinci
+DaVinci().UserAlgorithms += [dtt]
+DaVinci().InputType = 'DST'
+DaVinci().TupleFile = 'DVntuple.root'
+DaVinci().PrintFreq = 1000
+DaVinci().DataType = '2011'
```
Comment (betatim) [inline 09-minimal-dv-job.md]: How did you find out that 2011 is the one to use for this MC sample? Worth mentioning as that seems like a FAQ

**Comment by betatim** [comment]
If we add something about the 2011 thing I think this is ready to merge?

File: 09-minimal-dv-job.md (hunk 2 above)
Comment (alexpearce) [inline 09-minimal-dv-job.md]: If the MC DST is from 2012, this is wrong and should be `'2012'`.

**Comment by alexpearce** [comment]
Just so I'm sure then, the DST these options will run over is an MC DST? If that's the case we will need to mention DDDB and CondDB tags.

File: 09-minimal-dv-job.md (hunk 2 above)
Comment (betatim) [inline 09-minimal-dv-job.md]: It is. The way to know is to look at the BK path. Can we add that sentence somewhere?

**Comment by betatim** [comment]
The files I got for the book-keeping example are MC, full DST (I think).

**Comment by alexpearce** [comment]
OK I'll add something then, though it is a little fiddly unfortunately.

**Comment by betatim** [comment]
https://github.com/betatim/lhcb-first-analysis-steps/blob/gh-pages/data/MC_2012_27163003_Beam4000GeV2012MagDownNu2.5Pythia8_Sim08e_Digi13_Trig0x409f0045_Reco14a_Stripping20NoPrescalingFlagged_ALLSTREAMS.DST

We can take the info from this file no? Which also answers the question "How do I find my tags?"

**Comment by alexpearce** [comment]
Yes, we can. I've had instances where that information wasn't present in the bookkeeping information though, so you have to try [alternative procedures](https://wiki.physik.uzh.ch/lhcb/davinci:dbtags).

One method is to use the Dirac API, which often produces the same result as in the bookkeeping `.py` file:

``` shell
$ SetupLHCbDirac
$ lhcb-proxy-init
$ dirac-bookkeeping-production-informations 35742
```

Where 35742 is the job production ID, the number after the `ALLSTREAMS.DST` part of the LFN with the leading zeros removed. For me, today, this produces

```
Production Info:
    Configuration Name: MC
    Configuration Version: 2012
    Event type: 27163003
-----------------------
 StepName: Merge14 for Sim08
    ApplicationName    : LHCb
    ApplicationVersion : v35r4
    OptionFiles        : $APPCONFIGOPTS/Merging/CopyDST.py
    DDB                : None
    CONDDB             : None
    ExtraPackages      :AppConfig.v3r164
-----------------------
Number of Steps   10
Total number of files: 20
         LOG:10
         ALLSTREAMS.DST:10
Number of events
File Type           Number of events    Event Type          EventInputStat
ALLSTREAMS.DST      140687              27163003            140687
Path:  /MC/2012/Beam4000GeV-2012-MagDown-Nu2.5-Pythia8/Sim08e/Digi13/Trig0x409f0045/Reco14a/Stripping20NoPrescalingFlagged
/MC/2012/Beam4000GeV-2012-MagDown-Nu2.5-Pythia8/Sim08e/Digi13/Trig0x409f0045/Reco14a/Stripping20NoPrescalingFlagged/27163003/ALLSTREAMS.DST
```

This shows the information for the merging production step, where all the individual DSTs are merged, and has `None` for both tags, so is unhelpful.

Unfortunately, the only foolproof way I know of getting these tags is to
1. Find out the production ID by inspecting the LFN;
2. Go to the [transformations monitor](https://lhcb-web-dirac.cern.ch/DIRAC/LHCb-Production/lhcb_user/Production/LHCbTransformationMonitor/display) and input the production ID in the “ProductionID” search field;
3. Click on the row that's returned as the search result and then “Show Request”, which will show the MC production request that created that production ID;
4. If the request has subrequests (as it does in this case), click the little “+” icons on the left of the row to expand it, and click the “+” icons again until you find the request that corresponds to the event type you're interested in (ID 20789 here);
5. Click on the row with the correct event type and then click “View”;
6. The first step shown, the step that ran Gauss, will show the tags.

For data, [the advice](https://twiki.cern.ch/twiki/bin/view/LHCb/RecommendedTags) is to just use the latest tags, so you don't need to go through all of this. For simulation it really is necessary: if you don't, you can end up with the wrong magnet polarity value in your ntuple, amongst other things.

**Comment by alexpearce** [comment]
Pushed information on database tags in 6eb9a0d.

**Comment by betatim** [comment]
I see another little lesson on how to find DB tags coming ...

If you fix the year in the options file, I am :+1: on merging.

**Comment by alexpearce** [comment]
Done.

**Pull request merged**